src/decoration.coffee

```diff
@@ -46,6 +47,20 @@ class Decoration
     else
       type is decorationProperties.type
 
+  # Private: Check if the `decorationProperties.trueType` matches `type`
+  #
+  # * `decorationProperties` {Object} eg. `{type: 'line-number', class: 'my-new-class'}`
+  # * `type` {String} type like `'gutter'`, `'line'`, etc. `type` can also
+  #   be an {Array} of {String}s, where it will return true if the decoration's
+  #   type matches any in the array.
+  #
+  # Returns {Boolean}
+  @isTrueType: (decorationProperties, type) ->
```

Comment:
It makes sense that a `line-number` type could also count as a `gutter` type, but not necessarily vice-versa.
Right now, the translation from "OldToNewParams" would do the following:
* type: gutter  --> type: line-number, trueType: gutter, name: line-numbers
* type: line-numbers --> same as above
* type: gutter, name: 'custom-name' --> type: gutter, trueType: gutter, name: 'custom-name'